src/components/InsureeSearcher.js

```diff
@@ -154,47 +155,51 @@ class InsureeSearcher extends Component {
       (insuree) => insuree.chfId,
       (insuree) => insuree.lastName,
       (insuree) => insuree.otherNames,
-      (insuree) => (
+      this.isWorker? null : (insuree) => (
         <PublishedComponent
           pubRef="insuree.InsureeMaritalStatusPicker"
           withLabel={false}
           readOnly={true}
           value={insuree.marital || INSUREE_MARITAL_STATUS[0]}
         />
       ),
-      (insuree) => (
+        this.isWorker? null : (insuree) => (
         <PublishedComponent
           pubRef="insuree.InsureeGenderPicker"
           withLabel={false}
           readOnly={true}
           value={!!insuree.gender ? insuree.gender.code : null}
         />
       ),
-      (insuree) => insuree.email,
-      (insuree) => insuree.phone,
-      (insuree) => formatDateFromISO(this.props.modulesManager, this.props.intl, insuree.dob),
+        this.isWorker? null : (insuree) => insuree.email,
+        this.isWorker? null : (insuree) => insuree.phone,
+        this.isWorker? null : (insuree) => formatDateFromISO(this.props.modulesManager, this.props.intl, insuree.dob),
     ];
-    for (var i = 0; i < this.locationLevels; i++) {
-      // need a fixed variable to refer to as parentLocation argument
-      let j = i + 0;
-      formatters.push((insuree) =>
-        this.parentLocation(insuree.currentVillage || (!!insuree.family && insuree.family.location), j),
-      );
+    if (!this.isWorker) {
+        for (var i = 0; i < this.locationLevels; i++) {
```

Comment:
Can you precise more what does i' and 'j' mean? For example 'counter' etc?

Reply:
If this is not possible to find proper name for this variable, you can add explanation as comment.

Reply:
I can only suppose because I only wrapped the loop in the if statement. The loop is responsible for generating types of location different index means different location.